### ISSUE TITLE
`GitDumper` should also checkout the `main` branches

### ISSUE THREAD

**Issue by erikyao** [biothings/biothings.api#148]
When adding a new plugin to `pending.biothings.io`, the URL to the GitHub repo will be passed to `http://localhost:19080/dataplugin/register_url`, 

![image](https://user-images.githubusercontent.com/4813680/122142630-b8c45e00-ce04-11eb-8478-17bbe4d192b1.png)

which in the end calls the `AssistantManager.register_url()` method ([assistant.py#L699](https://github.com/biothings/biothings.api/blob/693f058b76869650abe608196891c963838c6390/biothings/hub/dataplugin/assistant.py#L699)).

The `AssistantManager` instance looks like to add a message (including the URL to register) to its corresponding MongoDB collection, and finally a `GitDumper` instance will receive the URL and then check the repo out. By default, `GitDumper` will only check `master` branches, but recently GitHub has changed its default branch name from `master` to `main`. Therefore our `GitDumper` cannot checkout the latest GitHub repo-based plugins.

The root cause in the code seems to be [dumper.py#L1072](https://github.com/biothings/biothings.api/blob/693f058b76869650abe608196891c963838c6390/biothings/hub/dataload/dumper.py#L1072):

```python
DEFAULT_BRANCH = "master"
```

**Comment by zcqian** [comment]
I think the better solution is to figure out what the remote HEAD is and use that as the default branch.

**Comment by newgene** [comment]
Per [this post](https://dev.to/bowmanjd/get-github-default-branch-from-the-command-line-powershell-or-bash-zsh-37m9), can use this to get the default git branch.

```
git ls-remote --symref "$REPO_URL" HEAD | sed -nE 's|^ref: refs/heads/(\S+)\s+HEAD|\1|p'
```

**Comment by zcqian** [comment]
The reason we don’t want to introduce `sed` is that there’s no guarantee that it exists, and piping probably doesn’t work as expected when running in a subprocess.

**Comment by newgene** [comment]
@zcqian certainly, within python code, you don't really need `sed`, should be easy to extract what you need with regex, same as what `sed` part intend to do in the above example.

**Comment by zcqian** [comment]
That’s what I’m thinking for now. We will invoke `git` and scrape its output for information, and in this case we will take the default branch in this order:
1. User defined default. The current class variable will default to None. Hoped this doesn’t break other people’s code.
2. Remote HEAD if only one remote is configured, raise if multiple remote is configured
3. master branch of the remote, with which the local repo master is set to track, if master exists
4. main if master does not exist 

In the long run I don’t like scraping the standard output of git, or using GitPython (we use this in some places, also probably scraping the output, given that it depends on the git command). We should look into solutions like libgit2.

**Comment by zcqian** [comment]
Fixed via #150
